__tests__/topics.test.js

```diff
@@ -0,0 +1,45 @@
+const app = require("../api/app");
+const request = require("supertest");
+const {
+  topicData,
+  userData,
+  articleData,
+  commentData,
+} = require("../db/data/test-data");
+const db = require("../db/connection");
+const seed = require("../db/seeds/seed");
+
+beforeEach(() => seed({ topicData, userData, articleData, commentData }));
+afterAll(() => db.end());
+
+describe("/api/topics", () => {
+  it("GET:200 sends an array to the client", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        expect(Array.isArray(body.topics)).toBe(true);
+      });
+  });
+
+  it("GET:200 return array is populated with objects that have the required properties - 'slug' and 'description'", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        body.topics.forEach((topic) => {
+          expect(topic).toHaveProperty("slug");
+          expect(topic).toHaveProperty("description");
+        });
+      });
+  });
+
+  it("GET:200 returns an array of the correct length", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.topics).toHaveLength(3);
```

Comment:
I would probably just add this check to the previous test block before the forEach as it is being used to validate the tests in the forEach

Reply:
Cool, thank you.